packages/camera/camera_platform_interface/lib/src/platform_interface/camera_platform.dart

```diff
@@ -0,0 +1,115 @@
+    _instance = instance;
+  }
+
+  /// Completes with a list of available cameras.
+  Future<List<CameraDescription>> availableCameras() {
+    throw UnimplementedError('availableCameras() is not implemented.');
+  }
+
+  /// Initializes the camera on the device and returns its textureId.
+  Future<int> initializeCamera(
+    CameraDescription cameraDescription, {
+    ResolutionPreset resolutionPreset,
+    bool enableAudio,
+  }) {
+    throw UnimplementedError('initializeCamera() is not implemented.');
+  }
+
+  /// Returns a Stream of [CameraEvent]s.
+  Stream<CameraEvent> cameraEventsFor(int textureId) {
+    throw UnimplementedError('videoEventsFor() has not been implemented.');
+  }
+
+  /// Captures an image and saves it to [path].
+  Future<void> takePicture(int textureId, String path) {
+    throw UnimplementedError('takePicture() is not implemented.');
+  }
+
+  /// Prepare the capture session for video recording.
+  Future<void> prepareForVideoRecording() {
+    throw UnimplementedError('prepareForVideoRecording() is not implemented.');
+  }
+
+  /// Start a video recording and save the file to [path].
+  ///
+  /// A path can for example be obtained using
+  /// [path_provider](https://pub.dartlang.org/packages/path_provider).
+  ///
+  /// The file is written on the flight as the video is being recorded.
+  /// If a file already exists at the provided path an error will be thrown.
+  /// The file can be read as soon as [stopVideoRecording] returns.
+  Future<void> startVideoRecording(int textureId, String path) {
+    throw UnimplementedError('startVideoRecording() is not implemented.');
+  }
+
+  /// Stop the video recording.
+  Future<void> stopVideoRecording(int textureId) {
+    throw UnimplementedError('stopVideoRecording() is not implemented.');
+  }
+
+  /// Pause video recording.
+  Future<void> pauseVideoRecording(int textureId) {
+    throw UnimplementedError('pauseVideoRecording() is not implemented.');
+  }
+
+  /// Resume video recording after pausing.
+  Future<void> resumeVideoRecording(int textureId) {
+    throw UnimplementedError('resumeVideoRecording() is not implemented.');
+  }
+
+  /// Start streaming images from platform camera.
+  ///
+  /// When running continuously with [CameraPreview] widget, this function runs
+  /// best with [ResolutionPreset.low]. Running on [ResolutionPreset.high] can
+  /// have significant frame rate drops for [CameraPreview] on lower end
+  /// devices.
+  // TODO(bmparr): Add settings for resolution and fps.
+  Future<void> startImageStream(onLatestImageAvailable onAvailable) {
+    throw UnimplementedError('startImageStream() is not implemented.');
+  }
+
+  /// Stop streaming images from platform camera.
+  Future<void> stopImageStream() {
+    throw UnimplementedError('stopImageStream() is not implemented.');
+  }
+
+  /// Releases the resources of this camera.
+  Future<void> dispose(int textureId) {
+    throw UnimplementedError('dispose() is not implemented.');
+  }
+}
```

Comment:
Note: there's one method missing in this platform interface; something that knows how to render a widget for the camera preview. See [`Widget buildView` in the Maps plugin](https://github.com/flutter/plugins/blob/master/packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/platform_interface/google_maps_flutter_platform.dart#L312-L318).

(Here's how the core maps plugin [calls the `buildView` method of the platform_interface](https://github.com/flutter/plugins/blob/master/packages/google_maps_flutter/google_maps_flutter/lib/src/google_map.dart#L237-L241) in its own `build` method! If you look at the method_channel and web implementations, you'll see how different they are!)

**Note:** Apologies for the `Map<String, dynamic> creationParams`, those should be expanded to actual constructor parameters, with semantic names, but creation params in maps has grown to be AN ENORMOUS OBJECT and I haven't had time to refactor that yet (Yet another problem of parameter bags haha :P)

Reply:
I really like this solution. I will make sure to incorporate it. For the camera plugin this should be pretty straight forward since we only have to worry about the `textureId` (maybe we should rename it to `cameraId`) parameter. 

Do I understand it correctly that in general the method_channel can simply return a `Texture` widget, while the web version will return a `HtmlElementView`?

Reply:
I think so, yes, the method_channel implementation will be pretty much the same that exists now within the build method of the CameraPreview widget, and the web version will necessarily have to return a HtmlElementView with a <video> element inside it, most likely!

Reply:
(This has been added as `Widget buildPreview(int cameraId)`)